src/freertos_drivers/spiffs/cc32x0sf/CC32x0SFSPIFFS.cxx

```diff
@@ -31,11 +31,64 @@
  * @date 1 January 2018
  */
 
+// #define LOGLEVEL INFO
+
+// This define is needed to call any ROM_xx function in the driverlib.
+#define USE_CC3220_ROM_DRV_API
+
+#include "utils/logging.h"
+
 #include "CC32x0SFSPIFFS.hxx"
 
 #include "spiffs.h"
 #include "inc/hw_types.h"
 #include "driverlib/flash.h"
+#include "driverlib/rom.h"
+#include "driverlib/cpu.h"
+
+/// Flash configuration register.
+#define FLASH_CONF 0x400FDFC8
+/// This bit in the FLASH_CONF register means that the banks are reversed by
+/// their address mapping.
+#define FCMME 0x40000000
+/// This value defines up to one bit that needs to be XOR-ed to the flash
+/// address before calling the flash APIs to cover for reversed flash banks.
+static const unsigned addr_mirror = (HWREG(FLASH_CONF) & FCMME) ? 0x80000 : 0;
+
+// Different options for what to set for flash write locking.
+
+// Global disable interrupts.
+//
+// #define DI() asm("cpsid i\n")
+// #define EI() asm("cpsie i\n")
+
+// Critical section (interrupts better than MIN_SYSCALL_PRIORITY are still
+// running).
+//
+// #define DI() portENTER_CRITICAL()
+// #define EI() portEXIT_CRITICAL()
+
+// Disable interrupts with a custom priority limit (must not be zero).
+//
+// unsigned ppri;
+// constexpr unsigned minpri = 0x40;
+// #define DI() ppri = CPUbasepriGet(); CPUbasepriSet(minpri); HWREG(FLASH_CONF) |=  0x20110000;
+// #define EI() CPUbasepriSet(ppri);
+
+// No write locking.
+#define DI()
+#define EI()
+
+// This ifdef decides whether we use the ROM or the flash based implementations
+// for Flash write and erase. It also supports correcting for the reversed bank
+// addresses.
+#if 1
```

Comment:
I presume you have hard coded the use of the ROM API to be enabled. Is the #else dead code then?

Reply:
We will not use the other code in production, but flipping these ifdefs around is helpful to debug and reproduce crashes around the flash.